beater/server.go

```diff
@@ -108,14 +117,16 @@ func authHandler(secretToken string, h http.Handler) http.Handler {
 	})
 }
 
-func appHandler(p processor.Processor, config Config, report reporter) http.Handler {
+func appHandler(constructor constructor, config Config, report reporter) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		code, err := processRequest(r, p, config.MaxUnzippedSize, report)
+		code, err := processRequest(r, constructor, config.MaxUnzippedSize, report)
 		sendStatus(w, r, code, err)
 	})
 }
 
-func processRequest(r *http.Request, p processor.Processor, maxSize int64, report reporter) (int, error) {
+func processRequest(r *http.Request, new constructor, maxSize int64, report reporter) (int, error) {
+
+	var p processor.Processor = new()
```

Comment:
I like this improvement, to have one dedicated processor per request. 
Currently we do call the [`CreateSchema`](https://github.com/elastic/apm-server/blob/master/processor/validator.go#L11) method though when creating a new processor. With your changes this would be done for every request. I don't think is the way to go. Could you update that the schema is created once per endpoint and then reused per request?

Reply:
good point! we totally should do that.
besides, this is not meant to be merged (at least just yet), this conflicts with PR227 and i'd like to see what everyone thinks of https://github.com/elastic/apm-server/issues/230

Reply:
I totally support this improvement, as I think it makes things clearer and less error prone.